pkg/sdk/validation/manifest/json_remote_type.go

```diff
@@ -82,6 +97,55 @@ func (v *RemoteTypeValidator) checkAdditionalRefs(ctx context.Context, entity ty
 	return res, nil
 }
 
+func (v *RemoteTypeValidator) validateBackendStorageSchema(ctx context.Context, entity types.Type) (ValidationResult, error) {
+	res := ValidationResult{}
+
+	if !entity.IsExtendingHubStorage() {
+		return res, nil
+	}
+
+	opts := []public.TypeOption{
+		public.WithTypeFilter(gqlpublicapi.TypeFilter{
+			PathPattern: ptr.String(types.GenericBackendStorageSchemaTypePath),
+		}),
+		public.WithTypeLatestRevision(public.TypeRevisionSpecFields),
+	}
+
+	gotTypes, err := v.hub.ListTypes(ctx, opts...)
+	if err != nil {
+		return res, errors.Wrap(err, "while fetching Types")
+	}
+
+	if len(gotTypes) == 0 || gotTypes[0].LatestRevision == nil || gotTypes[0].LatestRevision.Spec == nil {
+		return res, fmt.Errorf("cannot find generic backend storage schema")
+	}
+
+	storageBytes, err := json.Marshal(gotTypes[0].LatestRevision.Spec.JSONSchema)
+	if err != nil {
+		return res, errors.Wrap(err, "while marshaling generic backend storage schema to bytes")
+	}
+
+	var storageSchema string
+	err = json.Unmarshal(storageBytes, &storageSchema)
+	if err != nil {
+		return res, errors.Wrap(err, "while converting generic backend storage schema to string")
+	}
+
+	schemaLoader := gojsonschema.NewStringLoader(storageSchema)
+	dataLoader := gojsonschema.NewStringLoader(entity.Spec.JSONSchema.Value)
+
+	result, err := gojsonschema.Validate(schemaLoader, dataLoader)
+	if err != nil {
+		return res, errors.Wrap(err, "while validating JSON schema against the generic storage backend schema")
+	}
+
+	for _, err := range result.Errors() {
+		res.Errors = append(res.Errors, fmt.Errorf("while validating JSON Schema against the generic storage backend schema (`%s`):: %s", types.GenericBackendStorageSchemaTypePath, err.String()))
```

Comment:
```suggestion
		res.Errors = append(res.Errors, errors.Wrapf(err, "while validating JSON Schema against the generic storage backend schema (%q)"))
```

Reply:
I think we cannot use it as the `err` is the type of `ResultError` (not `error`) so there is a mismatch of the interface.

Reply:
Oh, you're right 👍  But please use `%q` instead of backticks and `%s` and remove double colon 🙂 

```suggestion
		res.Errors = append(res.Errors, fmt.Errorf("while validating JSON Schema against the generic storage backend schema (%q): %s", types.GenericBackendStorageSchemaTypePath, err.String()))
```

Reply:
Fixed